### PR TITLE
Add challenges

### PR DESCRIPTION
**Pull request by StefanJum**

### CODE REVIEW

**Comment by craciunoiuc** [comment]
The challenges look good, I'll have to also try them out, but the difficulty seems fine.

**Comment by craciunoiuc** [comment]
Considering the target audience, I would not call the second task `simple`, but there must be something to set people apart.
Or maybe I just misinterpreted the challenge.

The first and third challenge are good though.

**Comment by StefanJum** [comment]
> Considering the target audience, I would not call the second task `simple`, but there must be something to set people apart. Or maybe I just misinterpreted the challenge.
> 
> The first and third challenge are good though.

Makes sense. I can move it to no. 3 just to not hit the people trying too soon.
The challenge can be solved with `strings` or `ltrace` (it waits for a password, does a `strcmp` and then prints the flag), maybe I can make it more verbose, by adding some input prompts (e.g. `Enter the password:`) so it's clear for everyone what they should do.

I can also remove or replace it, I could not come up with an easier idea that would not be just too easy and I thought that leaving only the other 2 is a little too less work, but if this one is too difficult we can leave it out.

**Comment by craciunoiuc** [comment]
Ah whoops, okay. I tried gdb-ing inside it but for some reason it didn't work and then I saw the key and I thought they had to overwrite the strcmp bytes with nop operations 😅

I supposed the password was hashed.

Okay then you can leave it like this, but maybe hint that the password is already in the binary.

**Comment by StefanJum** [comment]
> Okay then you can leave it like this, but maybe hint that the password is already in the binary.

Yes, I'll do that.

File: 01-build-system/src/main.c [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include <stdio.h>
+#include <myheader.h>
+
+extern char* traveler;
```
Comment (RaduNichita) [inline 01-build-system/src/main.c]:
```suggestion
extern char *traveler;
```

File: 01-build-system/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+`-- some
+    `-- far
+        `-- away
+            |-- include
+            |   `-- myheader.h
+            `-- src
+                `-- values.c
+```
+
+You are asked to create a `Makefile` that will be used to build everything into an executable file.
```
Comment (RaduNichita) [inline 01-build-system/README.md]:
```suggestion
Your task is to create a `Makefile` that will be used to build everything into an executable file.
```

File: 01-build-system/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+After everything is done, if you run the executable you will find a flag:
+
+```console
+$ ./main
+[...]
+USoC{...}
+```
+
+Copy that flag and paste it in the [registration form](TODO), along with the `Makefile` you created.
```
Comment (RaduNichita) [inline 01-build-system/README.md]: Maybe they should create a zip archive in which they will upload multiple files?

File: 01-build-system/src/main.c [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+extern char *get_key();
+
+int main(int argc, char *argv[])
```
Comment (RaduNichita) [inline 01-build-system/src/main.c]:
```suggestion
int main(void)
```

File: 03-my-first-unikernel/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# 03-my-first-unikernel
+
+## Challenge Description
+
+You are given a unikernel image.
```
Comment (RaduNichita) [inline 03-my-first-unikernel/README.md]: Maybe put a link / add a short description about  what is a unikernel image?

File: 03-my-first-unikernel/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+Run it using [`qemu-system`](https://manpages.debian.org/jessie/qemu-system-x86/qemu-system-x86_64.1.en.html) and get the flag.
+
+If everything goes well, the output of the image will look something like this:
+
+```console
+$ your-run-command
```
Comment (RaduNichita) [inline 03-my-first-unikernel/README.md]:
```suggestion
$ <your_run_command>
```

**Review by RaduNichita — CHANGES_REQUESTED** [review]
Looking awesome, nice work @StefanJum !

I left a few nitpicks and little suggestions.

**Comment by StefanJum** [comment]
The challenges were published in the main [unikraft docs repository](https://github.com/unikraft/docs/tree/main/content/en/community/hackathons/sessions/registration-challenges)

**Pull request closed without merging**